python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java

```diff
@@ -55,6 +68,116 @@ void test() {
     System.out.println("hello");
   }
 
+  @Test
+  void testTypeshedImports() {
+    FileInput root = inferTypes("""
+      import cryptography.hazmat.primitives.asymmetric
+      import datetime
+      a = datetime.date(year=2023, month=12, day=1)
+      """);
+
+    var importName = (ImportName) root.statements().statements().get(0);
```

Comment:
Nice job on the test refactoring, this looks much cleaner and maintainable.

I think we should remove the base `test` that relies on the `script.py` file though (assuming we've migrated all of its use cases to other tests). 
In this PR we could leave it there as a reference if not all cases have dedicated tests, but we shouldn't forget to remove it later.